addons/account/models/res_partner_bank.py

```diff
@@ -86,7 +86,11 @@ def _compute_duplicate_bank_partner_ids(self):
             ids=self.ids,
         )))
         for bank in self:
-            bank.duplicate_bank_partner_ids = self.env['res.partner'].browse(id2duplicates.get(bank._origin.id))
+            duplicate_record = id2duplicates.get(bank._origin.id)
+            if duplicate_record and duplicate_record[0]:
+                bank.duplicate_bank_partner_ids = self.env['res.partner'].browse(duplicate_record)
+            else:
+                bank.duplicate_bank_partner_ids = []
```

Comment:
```py
                bank.duplicate_bank_partner_ids = []
```

A many2many field with an empty list value?? ... No?!?  😖 
The problem is `res.partner(None,)` because of how is handled in RecordSnapshot
There should be something that avoids having `None` as `id`.

<img width="1920" height="1040" alt="image" src="https://github.com/user-attachments/assets/9f434765-b0ad-4487-8cd3-a08bc7e98c4a" />
(I have added the bit highlighted in red locally, it's not there in `saas-18.4`)

If you want to solve this locally, just turning `duplicate_record` from `[None]` to `[]` just works. It will turn it into the empty recordset `res.partner()` and it won't explode (I mean, traceback).

```suggestion
        for bank in self:
            duplicate_record = id2duplicates.get(bank._origin.id) or []
            duplicate_record = [x for x in duplicate_record if x]
            bank.duplicate_bank_partner_ids = self.env['res.partner'].browse(duplicate_record)
```

(As proof for the bug being in `models.py`, a simple call to `self.env['res.partner'].browse([None]).ids` will show that it's equal to `[]`, so this recordset is kinda handled well by the system.)